connectors/s3-connector/src/main/scala/zio/connect/s3/LiveS3Connector.scala

```diff
@@ -14,127 +14,185 @@ import zio.aws.s3.model.{
+        getS3(region).flatMap(
+          _.deleteObjects(
+            DeleteObjectsRequest(
+              bucket = AwsBucketName(bucketName),
+              delete = Delete(objects = objectKeys.map(a => ObjectIdentifier(AwsObjectKey(a))))
+            )
+          ).mapError(a => S3Exception(a.toThrowable))
         )
       }
-      .mapError(a => S3Exception(a.toThrowable))
 
-  override def getObject(bucketName: => BucketName, key: => ObjectKey)(implicit
+  override def getObject(bucketName: => BucketName, key: => ObjectKey, region: => Region)(implicit
     trace: Trace
   ): ZStream[Any, S3Exception, Byte] =
-    ZStream
-      .unwrap(
-        s3.getObject(GetObjectRequest(bucket = AwsBucketName(bucketName), key = AwsObjectKey(key)))
-          .map(a => a.output)
-      )
-      .mapError(a => S3Exception(a.toThrowable))
-
-  override def listBuckets(implicit
+    ZStream.unwrap(getS3(region).flatMap { s3 =>
+      s3.getObject(GetObjectRequest(bucket = AwsBucketName(bucketName), key = AwsObjectKey(key)))
+        .mapError(a => S3Exception(a.toThrowable))
+        .flatMap(o => ZIO.succeed(o.output))
+        .map((r: ZStream[Any, AwsError, Byte]) => r.mapError(a => S3Exception(a.toThrowable)))
+    })
+
+  override def listBuckets(region: => Region)(implicit
     trace: Trace
   ): ZStream[Any, S3Exception, BucketName] =
-    ZStream
-      .fromIterableZIO(
-        s3.listBuckets().flatMap(_.getBuckets.map(_.flatMap(_.name.toList.map(BucketName(_)))))
-      )
-      .mapError(a => S3Exception(a.toThrowable))
+    ZStream.fromIterableZIO {
+      for {
+        s3          <- getS3(region)
+        listBuckets <- s3.listBuckets().mapError(a => S3Exception(a.toThrowable))
+        buckets     <- listBuckets.getBuckets.mapError(a => S3Exception(a.toThrowable))
+        bucketNames <- ZIO.succeed(buckets.flatMap(b => b.name.toList.map(BucketName(_))))
+
+      } yield bucketNames
+    }
 
-  override def listObjects(bucketName: => BucketName)(implicit
+  override def listObjects(bucketName: => BucketName, region: => Region)(implicit
     trace: Trace
   ): ZStream[Any, S3Exception, ObjectKey] =
-    ZStream
-      .fromIterableZIO(
-        s3.listObjects(ListObjectsRequest(bucket = AwsBucketName(bucketName)))
-          .map(_.contents.map(_.flatMap(_.key.toChunk.map(ObjectKey(_)))).getOrElse(Chunk.empty[ObjectKey]))
-      )
-      .mapError(a => S3Exception(a.toThrowable))
-
-  override def moveObject(implicit
+    ZStream.fromIterableZIO {
+      for {
+        s3 <- getS3(region)
+        objects <- s3.listObjects(ListObjectsRequest(bucket = AwsBucketName(bucketName)))
+                     .mapBoth(
+                       a => S3Exception(a.toThrowable),
+                       _.contents.map(_.flatMap(_.key.toChunk.map(ObjectKey(_)))).getOrElse(Chunk.empty[ObjectKey])
+                     )
+      } yield objects
+
+    }
+
+  override def moveObject(region: => Region)(implicit
```

Comment:
I think this can be implemented in the trait based on the moveObject(source, target)